official/keras_application_models/benchmark_main.py

```diff
@@ -0,0 +1,242 @@
+from official.utils.flags import core as flags_core
+from official.utils.logs import logger
+
+_NUM_CHANNELS = 3
+_NUM_CLASSES = 1000
+
+# Define a dictionary that maps model names to their model classes inside Keras
+MODELS = {
+    "vgg16": tf.keras.applications.VGG16,
+    "vgg19": tf.keras.applications.VGG19,
+    "inception": tf.keras.applications.InceptionV3,
+    "xception": tf.keras.applications.Xception,
+    "resnet": tf.keras.applications.ResNet50,
+    "inceptionresnet": tf.keras.applications.InceptionResNetV2,
+    "mobilenet": tf.keras.applications.MobileNet,
+    "densenet121": tf.keras.applications.DenseNet121,
+    "densenet169": tf.keras.applications.DenseNet169,
+    "densenet201": tf.keras.applications.DenseNet201,
+    "nasnetlarge": tf.keras.applications.NASNetLarge,
+    "nasnetmobile": tf.keras.applications.NASNetMobile,
+}
+
+
+def get_model_callbacks(benchmark_level, callbacks_list, batch_size=None,
+                        epoch_size=None, metrics=None, metric_logger=None):
+  """Create a list of callbacks.
+
+  Args:
+    benchmark_level: a string to indicate if the benchmark is `batch_based` or
+      `epoch_based`. Check `model_callbacks.py` for more details.
+    callbacks_list: a list of strings to name desired callbacks. Allowed:
+      ExamplesPerSecondCallback, LoggingMetricCallback, which are defined
+      in CALLBACKS.
+    batch_size: an int of batch size.
+    epoch_size: an int of epoch size, which is the total number of training
+      samples in one epoch.
+    metrics: `dict` that maps string-valued tags to metric names, or `iterable`
+      of metric names.
+    metric_logger: instance of `BenchmarkLogger`, the benchmark logger that
+      callback should use to write the log.
+
+  Returns:
+    callbacks: a list of callbacks.
+
+  Raises:
+    ValueError: if an unrecognized callback name is passed.
+  """
+  if not callbacks_list:
+    return []
+
+  batch_based = True if benchmark_level == "batch_based" else None
+  epoch_based = True if benchmark_level == "epoch_based" else None
+
+  callbacks = []
+  for callback in callbacks_list:
+    callback_name = callback.strip().lower()
+    if callback_name not in model_callbacks.CALLBACKS:
+      raise ValueError("Unrecognized callback requested: {}".format(callback))
+
+    if callback_name == model_callbacks.CALLBACKS[0]:
+      exp_per_second_callback = model_callbacks.ExamplesPerSecondCallback(
+          batch_size=batch_size, epoch_size=epoch_size,
+          batch_based=batch_based, epoch_based=epoch_based,
+          metric_logger=metric_logger)
+      callbacks.append(exp_per_second_callback)
+    else:
+      logging_metric_callback = model_callbacks.LoggingMetricCallback(
+          metrics=metrics, metric_logger=metric_logger,
+          batch_based=batch_based, epoch_based=epoch_based)
+      callbacks.append(logging_metric_callback)
+
+  return callbacks
+
+
+def generate_synthetic_input_dataset(image_size, batch_size):
```

Comment:
We have used this function elsewhere-- move to a util in model_helpers that takes two lists of ints for input shapes?